tests/test_auditor.py

```diff
@@ -0,0 +1,17 @@
+import pytest
+
+from auditor.auditor import retry
+
+
+def test_retry():
+
+    class CustomError(Exception):
+        pass
+
+    def inner_retry():
+        if 4 % 2 == 0:
+            #: If this exception gets raised, it means retry() has either not been called or has gone through all it's tries and has raised the original exception.
+            raise CustomError
+
+    with pytest.raises(CustomError):
+        retry(inner_retry)
```

Comment:
**suggestion**: You could use the mock library to assert that `inner_retry` was called a certain number of times.

Reply:
Yeah, I saw those methods, I'm still trying to wrap my head around mock and how to implement it with my current code structure. I may need to do some refactoring before it makes sense.